packages/window-size-tool/src/engine.js

```diff
@@ -123,8 +122,23 @@ function checkRule(node, rule, verbose = true) {
         );
       }
     }
+    if (pass == true){
+        pass = false;
+        rule.allTerms.forEach((term) => {
+        if (!pass) {
+          const result2 = xp.select(term, node);
+          pass = pass || result2;
```

Comment:
In the scenario where `allTerms` has more than one term, then the first time this loop runs, `pass` starts out `false`, so it will get reassigned to the value of `result2` . Let’s assume the first term returns true so now `pass` is also `true`.

On the second loop, because `pass` is now `true` the test on line 128 `if (!pass) {` won’t pass and thus we’ll never check the second term and the rule **will** get applied. This is a problem because the second term might have returned false and thus the rule **shouldn't** apply.

```suggestion
if (pass == true){
    rule.allTerms.forEach((term) => {
        if (pass) {
            const result2 = xp.select(term, node);
            pass = result2;
            if (verbose) {
            console.debug(
              ` term:  ${
                result2 ? chalk.green(`${result2} `) : chalk.red(result2)
              } <- [${chalk.magenta(term)}]`
            );
          }
        }
  });
}
```

Reply:
Yes, you’re right the problem was on line 128, changing `if (!pass) {` to` if (pass) {` and assigning the value of `result2` to  `pass` worked! 

Thanks Graham!